filebeat/input/filestream/internal/input-logfile/manager.go

```diff
@@ -192,7 +187,15 @@ func (cim *InputManager) Create(config *common.Config) (input.Input, error) {
 	defer pStore.Release()
 
 	prospectorStore := newSourceStore(pStore, sourceIdentifier)
-	err = prospector.Init(prospectorStore)
+
+	// create a store with the old default ID
+	globalIdentifier, err := newSourceIdentifier(cim.Type, "")
+	if err != nil {
+		return nil, fmt.Errorf("cannot create global identifier for input: %w", err)
+	}
+	globalStore := newSourceStore(pStore, globalIdentifier)
+
+	err = prospector.Init(prospectorStore, globalStore, sourceIdentifier.ID)
```

Comment:
Why are you cleaning during prospector initialization? I think we can do it earlier when the manager is initialized: https://github.com/elastic/beats/blob/main/filebeat/input/filestream/internal/input-logfile/manager.go#L94 or https://github.com/elastic/beats/blob/main/filebeat/input/filestream/internal/input-logfile/manager.go#L118

Reply:
Because that is where the registry is cleaned and also the "update identifiers" runs. It looked like a similar operation to me. Also, most of the data needed to implement this logic is already available there.

I'll look into those other places you mentioned as well.

Reply:
I took a look at those methods, but they do not have access to the input IDs, now the "configuration yaml". The first place I saw with access to the input IDs is the `Create` method: https://github.com/elastic/beats/blob/9e6536d964f0135940b9c416bcb1c0a41af99478/filebeat/input/filestream/internal/input-logfile/manager.go#L157-L169


But I also need the files information from the prospector, so the `prospector.Init` still looks like the best place for me.

Reply:
> Because that is where the registry is cleaned and also the "update identifiers" runs.

Updating identifiers is the responsibility of the prospector. It should only update state identifiers if it owns the given states. However, cleaning up legacy states does not belong to any specific prospector. It is the responsibility of the filestream manager to clean and maintain all filestream states.

> they do not have access to the input IDs

Why do we need access to input IDs? We know that we have to clean up the states that belong to `.global` ID.

Reply:
Nevermind, I've reread you code. You are correct.